File: app/src/main/java/me/devsaki/hentoid/parsers/images/ParseProgress.java
package me.devsaki.hentoid.parsers.images;

import java.util.concurrent.atomic.AtomicBoolean;

import me.devsaki.hentoid.parsers.ParseHelper;

class ParseProgress {

    private long contentId;
    private long storedId;
    private int currentStep;
    private int maxSteps;
    private boolean hasStarted = false;
    private final AtomicBoolean processHalted = new AtomicBoolean(false);

    void start(long contentId, long storedId, int maxSteps) {
        this.contentId = contentId;
        this.storedId = storedId;
        currentStep = 0;
        this.maxSteps = maxSteps;
        ParseHelper.signalProgress(contentId, storedId, currentStep, maxSteps);
        hasStarted = true;
    }

    boolean hasStarted() {
        return hasStarted;
    }

    boolean isProcessHalted() {
        return processHalted.get();
    }

    void haltProcess() {
        processHalted.set(true);
    }

    void advance() {
        ParseHelper.signalProgress(contentId, storedId, ++currentStep, maxSteps);
    }

    void complete() {
        ParseHelper.signalProgress(contentId, storedId, maxSteps, maxSteps);
    }
}

File: app/src/main/java/me/devsaki/hentoid/parsers/images/ParseProgress.kt
package me.devsaki.hentoid.parsers.images

import me.devsaki.hentoid.parsers.ParseHelper
import java.util.concurrent.atomic.AtomicBoolean

class ParseProgress {
    private var contentId: Long = 0
    private var storedId: Long = 0
    private var currentStep = 0
    private var maxSteps = 0
    private var hasStarted = false
    private val processHalted = AtomicBoolean(false)

    fun start(contentId: Long, storedId: Long, maxSteps: Int) {
        this.contentId = contentId
        this.storedId = storedId
        currentStep = 0
        this.maxSteps = maxSteps
        ParseHelper.signalProgress(contentId, storedId, currentStep, maxSteps)
        hasStarted = true
    }

    fun hasStarted(): Boolean {
        return hasStarted
    }

    fun isProcessHalted(): Boolean {
        return processHalted.get()
    }

    fun haltProcess() {
        processHalted.set(true)
    }

    fun advance() {
        ParseHelper.signalProgress(contentId, storedId, ++currentStep, maxSteps)
    }

    fun complete() {
        ParseHelper.signalProgress(contentId, storedId, maxSteps, maxSteps)
    }
}